### ISSUE TITLE
What does the cosine_scale mean?

### ISSUE THREAD

**Issue by Jianghao-Li** [muliyangm/DeCa#1]
Why multiply a self.scale in front of the predicted probability? Because before CE loss the probability will pass a softmax, i think the scale is meaningless, why is the result so much worse when I set the scale equal to 1?

**Comment by muliyangm** [comment]
Thank you for your interest in our work. Please note that the cosine scale is actually the reciprocal of the softmax temperature parameter [1], which controls the sharpness of the output distribution after softmax, and subsequently affects the loss value. A temperature parameter (which should be less than 1, eg, 0.1) is often required when cosine similarity (instead of dot product) is used for calculating the logits, because cosine similarity ignores the magnitude information and thus the original output distribution after softmax may become too flat (which is your case when setting the scale to 1), resulting in the low efficiency when optimizing with CE loss.

[1] Distilling the Knowledge in a Neural Network, 2015

**Comment by Jianghao-Li** [comment]
Thanks for your reply, I understand why this scale is set, but is there any tips about how to set the specific value?

**Comment by muliyangm** [comment]
In my experience, the optimal values for this temperature parameter are for most time around 0.1-0.01. You can try some interpolations and you may find the tendency.
